### PR TITLE
tests: add ips midstream exception policy tests - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Add tests for the exception policy for midstream streams (https://redmine.openinfosecfoundation.org/issues/5468)

Previous PR: #907 

exception-policy-midstream-01: tests `pass-flow` behavior with midstream SYNACK
exception-policy-midstream-02: tests `drop-flow` behavior with midstream SYNACK
exception-policy-midstream-03: tests the default behavior with midstream SYNACK
exception-policy-midstream-04: tests the `pass-flow` behavior with midstream ACK
exception-policy-midstream-05: tests the `bypass` behavior with midstream SYNACK
exception-policy-midstream-06: tests `drop-flow` behavior with midstream ACK
exception-policy-midstream-07: tests `bypass` behavior with midstream ACK

Tests 06 and 07 are currently failing, I think this is related to the behavior described in 
https://redmine.openinfosecfoundation.org/issues/5510, still to be investigated.

### CODE REVIEW

File: tests/exception-policy-midstream-01/test.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      count: 0
+      match:
+        event_type: alert
+  - filter:
+      count: 1
+      match:
+        event_type: flow
+        flow.action: pass
+  - filter:
+      count: 1
```
Comment (victorjulien) [inline tests/exception-policy-midstream-01/test.yaml:16]: midstream is not enabled, so we should not have a http record

File: tests/exception-policy-midstream-04/test.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      alert.signature_id: 1
+- filter:
+    count: 1
+    match:
+      event_type: flow
+      flow.action: pass
+- filter:
+    count: 1
+    match:
+      event_type: http
```
Comment (victorjulien) [inline tests/exception-policy-midstream-04/test.yaml:19]: midstream is not enabled so we should have no http

File: tests/exception-policy-midstream-05/test.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+checks:
+  - filter:
+      count: 0
+      match:
+        event_type: alert
+  - filter:
+      count: 1
+      match:
+        event_type: flow
+        flow.action: pass
```
Comment (victorjulien) [inline tests/exception-policy-midstream-05/test.yaml:14]: does the flow have something to indicate it was bypassed?

**Review by victorjulien — CHANGES_REQUESTED** [review]
Think this needs
```diff
diff --git a/tests/exception-policy-midstream-01/test.yaml b/tests/exception-policy-midstream-01/test.yaml
index 9cd6c21..77d4e07 100644
--- a/tests/exception-policy-midstream-01/test.yaml
+++ b/tests/exception-policy-midstream-01/test.yaml
@@ -13,6 +13,6 @@ checks:
         event_type: flow
         flow.action: pass
   - filter:
-      count: 1
+      count: 0
       match:
         event_type: http
diff --git a/tests/exception-policy-midstream-04/test.yaml b/tests/exception-policy-midstream-04/test.yaml
index afc6c9d..e58e756 100644
--- a/tests/exception-policy-midstream-04/test.yaml
+++ b/tests/exception-policy-midstream-04/test.yaml
@@ -14,6 +14,6 @@ checks:
       event_type: flow
       flow.action: pass
 - filter:
-    count: 1
+    count: 0
     match:
       event_type: http
```

File: tests/exception-policy-midstream-05/test.yaml (hunk 3 above)
Comment (jufajardini) [reply tests/exception-policy-midstream-05/test.yaml:14]: yes, sorry, I'll add a check for the flow state.

**Comment by jufajardini** [comment]
Thanks, I think I still need to better wrap this whole concept in my head...

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata-verify/pull/921

**Pull request closed without merging**